distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -277,82 +280,119 @@ public void testLibffiEnv() throws Exception {
     }
 
     public void testTempDir() throws Exception {
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(tmpDir.toString(), Files.exists(tmpDir), is(true));
-            assertThat(tmpDir.getFileName().toString(), startsWith("elasticsearch-"));
-            return new ArrayList<>();
-        };
-        runForeground();
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), Files.exists(tmpDir), is(true));
+        assertThat(tmpDir.getFileName().toString(), startsWith("elasticsearch-"));
     }
 
     public void testTempDirWindows() throws Exception {
         Path baseTmpDir = createTempDir();
         sysprops.put("os.name", "Windows 10");
         sysprops.put("java.io.tmpdir", baseTmpDir.toString());
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(tmpDir.toString(), Files.exists(tmpDir), is(true));
-            assertThat(tmpDir.getFileName().toString(), equalTo("elasticsearch"));
-            assertThat(tmpDir.getParent().toString(), equalTo(baseTmpDir.toString()));
-            return new ArrayList<>();
-        };
-        runForeground();
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), Files.exists(tmpDir), is(true));
+        assertThat(tmpDir.getFileName().toString(), equalTo("elasticsearch"));
+        assertThat(tmpDir.getParent().toString(), equalTo(baseTmpDir.toString()));
     }
 
     public void testTempDirOverride() throws Exception {
         Path customTmpDir = createTempDir();
         envVars.put("ES_TMPDIR", customTmpDir.toString());
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
-            return new ArrayList<>();
-        };
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
+
         processValidator = pb -> assertThat(pb.environment(), not(hasKey("ES_TMPDIR")));
         runForeground();
     }
 
-    public void testTempDirOverrideMissing() throws Exception {
+    public void testTempDirOverrideMissing() {
         Path baseDir = createTempDir();
         envVars.put("ES_TMPDIR", baseDir.resolve("dne").toString());
-        var e = expectThrows(UserException.class, () -> runForeground());
+        var e = expectThrows(UserException.class, () -> ServerProcessUtils.setupTempDir(createProcessInfo()));
```

Comment:
nit: Should these temp dir tests move to a new ServerProcessUtilsTests?

Reply:
Good idea, moved (and split one of the tests)